### PR TITLE
fix(wallet): fix provider down event happening too often

### LINKED ISSUE #5555: Fix blockchain RPC providers down event happening too often
# Problem

POKT/Infura provider is down banner happens pretty often, which may cause issues with getting routes, sending tx, and overall spoil user experience

# Implementation

Research what causes it. Suspects:
- context cancellation for some commands can cause commands finishing with error "context cancelled" not handled properly
- reaching RPC limits per acc/total per day for history requests causing context cancellations as well
- missing trie node errors handling (archival blocks not available)

# Acceptance Criteria

#POKT/Infura provider is down" banner seen less often

### PR DESCRIPTION
**Pull request by IvanBelyakoff**
- handle context cancelled error
- do not count expected errors when calling tokenURI as providers errors
- use archival not for optimism (was silently added by Grove making the old URL non-archival)

Closes #5555

### CODE REVIEW

**Review by belalshehab — APPROVED** [review]

File: api/default_networks.go [hunk 1]
```diff
@@ -81,7 +81,7 @@ var sepolia = params.Network{
 var optimism = params.Network{
 	ChainID:                optimismChainID,
 	ChainName:              "Optimism",
-	RPCURL:                 "https://optimism-mainnet.rpc.grove.city/v1/",
+	RPCURL:                 "https://optimism-archival.rpc.grove.city/v1/",
```
Comment (dlipicar) [inline api/default_networks.go:84]: I know we needed client-side changes for changing RPC urls at some point. Is that no longer the case?

Comment (IvanBelyakoff) [reply api/default_networks.go:84]: I have a waiting code for desktop. Just dont want to rebase etc. Will make a PR once this is approved/merged

File: circuitbreaker/circuit_breaker_test.go [hunk 2]
```diff
@@ -45,20 +45,20 @@ func TestCircuitBreaker_ExecuteMultipleFallbacksFail(t *testing.T) {
 		ErrorPercentThreshold:  10,
 	})
 
-	circuitName := ""
+	circuitName := fmt.Sprintf("ExecuteMultipleFallbacksFail_%d", time.Now().Nanosecond()) // unique name to avoid conflicts with go tests `-count` option
```
Comment (dlipicar) [inline circuitbreaker/circuit_breaker_test.go:48]: nit: the probabilities are probably low enough as it is, but we most likely want to use `uuid` for "truly unique" identifiers

Comment (IvanBelyakoff) [reply circuitbreaker/circuit_breaker_test.go:48]: might be. I'll keep that in mind, thanx, for future. Noone actually does "--count=100" on CI AFAIK, that is for us mostly to verify that tests are not flaky

**Review by dlipicar — APPROVED** [review]

**Comment by igor-sirotin** [comment]
diff-coverage is an issue because of https://github.com/status-im/status-go/issues/5562. Merging.

**Pull request merged**